### PR TITLE
Add concurrent execution model test

### PR DESCRIPTION
**Pull request by arikmaor**
Should be merged right after https://github.com/binaris/marathon/pull/149

### CODE REVIEW

File: test/spec.yml [hunk 1]
```diff
@@ -457,6 +457,47 @@
         exit: 1
         err: "Empty string env 'EMPTY' in binaris.yml is not supported"
 
+- test: "concurrent execution model: deploy invoke multiple times"
+  steps:
+    -   in: |-
+            cat <<EOF >/home/dockeruser/test/binaris.yml
```
Comment (rylandg) [inline test/spec.yml]: I'm not sure why we started writing tests with the full absolute path but these tests run from the `test` directory and do not need it. Is there an advantage to this method?

Comment (bergundy) [reply test/spec.yml]: Good point @rylandg

File: test/spec.yml [hunk 2]
```diff
@@ -457,6 +457,47 @@
...
+              }
+            }
+            EOF
+    -   in: bn deploy concurrentfunc
+        out: |-
+            Deployed function concurrentfunc
+            Invoke with one of:
+              "bn invoke concurrentfunc"
+              "curl -H X-Binaris-Api-Key:* https://*"
+    -   in: "(for i in `seq 1 12`; do bn invoke concurrentfunc & done) | grep true"
```
Comment (bergundy) [inline test/spec.yml]: Why 12?

File: test/spec.yml [hunk 3]
```diff
@@ -457,6 +457,47 @@
...
+                executionModel: concurrent
+            EOF
+    -   in: |-
+            cat <<EOF >/home/dockeruser/test/function.js
+            function msleep(ms) {
+              return new Promise((resolve) => {
+                  setTimeout(resolve, ms);
+              });
+            }
+            let duringFlight = false;
```
Comment (bergundy) [inline test/spec.yml]: Maybe sleep always and hold a counter of inflight requests?
Would be nice to know how much concurrency we actually achieve

File: test/spec.yml [hunk 4]
```diff
@@ -884,7 +925,7 @@
       - npm i shelljs
       - |-
         cat <<EOF >/home/dockeruser/test/function.js
-        
+
```
Comment (bergundy) [inline test/spec.yml]: It's better to separate white space fixes and non related changes from PRs that change or add functionality.

File: test/spec.yml (hunk 2 above)
Comment (arikmaor) [reply test/spec.yml]: we were a bit tired when deciding on 12, we actually need `minBolts+1` so in the `testing` realm, that would be 3

File: test/spec.yml (hunk 3 above)
Comment (arikmaor) [reply test/spec.yml]: it would be much difficult to test, now I can just do ` | grep true`

File: test/spec.yml (hunk 4 above)
Comment (arikmaor) [reply test/spec.yml]: agree

**Pull request merged**